### ISSUE TITLE
2.2.0 fails to build with tesseract 3.02.01

### ISSUE THREAD

**Issue by nijel** [sirfz/tesserocr#53]
The compilation on slightly older system (Ubuntu Precise used on Travis CI) fails:

```
  Supporting tesseract v3.04.01

  Configs from pkg-config: {'libraries': ['lept', 'tesseract'], 'cython_compile_time_env': {'TESSERACT_VERSION': 197633}, 'library_dirs': ['/home/travis/build/WeblateOrg/weblate/.tesseract/lib'], 'include_dirs': ['/home/travis/build/WeblateOrg/weblate/.tesseract/include']}

  running bdist_wheel

  running build

  running build_ext

  building 'tesserocr' extension

  creating build

  creating build/temp.linux-x86_64-2.7

  gcc -pthread -fno-strict-aliasing -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/travis/build/WeblateOrg/weblate/.tesseract/include -I/opt/python/2.7.9/include/python2.7 -c tesserocr.cpp -o build/temp.linux-x86_64-2.7/tesserocr.o -std=c++11

  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for Ada/C/ObjC but not for C++ [enabled by default]

  cc1plus: error: unrecognized command line option ‘-std=c++11’

  error: command 'gcc' failed with exit status 1
```

The problem seems to be in [setup.py line 130](https://github.com/sirfz/tesserocr/blob/9c7774e2435dbcde5101c3550c86b702550213cc/setup.py#L130) as that compares version which is comparing TESSERACT_VERSION which is `197633` at this point with `4.` while it should probably compare with `0x40000`.

**Comment by sirfz** [comment]
Hello @nijel, as stated in the [requirements](https://github.com/sirfz/tesserocr#requirements), the least tesseract version supported is 3.04 and up.

**Comment by sirfz** [comment]
However, the line you referred to is indeed a bug. It should be `>= 0x040000`, so I will be fixing that.

**Comment by nijel** [comment]
I'm using separately compiled 3.04.01 (as can be seen from the logs), so that should be fine. The only problem is that compiler there doesn't understand c++11...

**Comment by sirfz** [comment]
Yes, just pushed a fix and will release shortly. Thanks for pointing it out :)

**Comment by sirfz** [comment]
Would appreciate it if you can test it in your environment (master branch).

**Comment by nijel** [comment]
I've just pushed it to separate branch, Travis build is here: https://travis-ci.org/WeblateOrg/weblate/builds/238065553

**Comment by sirfz** [comment]
The setup failed but for a different reason (`gcc: error: tesserocr.cpp: No such file or directory`), for some reason it didn't cythonize `tesserocr.pyx`. I already released v2.2.1 with the fix, could you try that instead? (might need to `pip install -U pip` in case it's not already up-to-date).

**Comment by nijel** [comment]
Thanks, it now builds just fine (the build is still in progress: https://travis-ci.org/WeblateOrg/weblate/builds/238088463).
